polaris/hub/client.py

```diff
@@ -287,17 +288,42 @@ def list_datasets(self, limit: int = 100, offset: int = 0) -> list[str]:
             A list of dataset names in the format `owner/dataset_name`.
         """
         with ProgressIndicator(
-            start_msg="Fetching artifacts...",
-            success_msg="Fetched artifacts.",
+            start_msg="Fetching datasets...",
+            success_msg="Fetched datasets.",
             error_msg="Failed to fetch datasets.",
         ):
-            response = self._base_request_to_hub(
-                url="/v1/dataset", method="GET", params={"limit": limit, "offset": offset}
+            # Step 1: Fetch enough v2 datasets to cover the offset and limit
+            v2_response = self._base_request_to_hub(
+                url="/v2/dataset", method="GET", params={"limit": limit + offset, "offset": 0}
             )
-            response_data = response.json()
-            dataset_list = [bm["artifactId"] for bm in response_data["data"]]
+            v2_data = v2_response.json().get("data", [])
+            v2_datasets = [dataset["artifactId"] for dataset in v2_data]
+
+            # Apply offset and limit to v2 datasets
+            v2_datasets_offset = v2_datasets[offset : offset + limit]
```

Comment:
**Suggestion:** This shouldn't be needed! The API endpoint will already limit and offset the results

Reply:
I made some tweaks.

Reply:
Good catch! I made some tweaks.